### PR TITLE
CollabDB initialisation

### PR DESCRIPTION
**Pull request by Daviancold**
**Resolves**
- #151 
- #145

### CODE REVIEW

File: backend/collaboration-service/src/controllers/collab.controller.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { Request, Response } from 'express'
+import { ITypedBodyRequest } from '@repo/request-types'
+import { CollabDto } from '../types/CollabDto'
+import { createSession, getSessionById } from '../models/collab.repository'
+
+export async function createSessionRequest(request: ITypedBodyRequest<CollabDto>, response: Response): Promise<void> {
+    const collabDto = CollabDto.fromRequest(request)
+    const errors = await collabDto.validate()
+    if (errors.length) {
+        const errorMessages = errors.map((error: ValidationError) => `INVALID_${error.property.toUpperCase()}`)
```
Comment (glemenneo) [inline backend/collaboration-service/src/controllers/collab.controller.ts:11]: Can refer to my lastest pr for how to send back verbose error messages

File: backend/collaboration-service/src/models/collab.model.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    chatHistory: {
+        type: [chatModelSchema],
+        required: true,
+    },
+    createdAt: {
+        type: Date,
+        required: true,
+    },
+})
+
```
Comment (glemenneo) [inline backend/collaboration-service/src/models/collab.model.ts:29]: Missing some indexes, will need for pagination

**Review by glemenneo — CHANGES_REQUESTED** [review]

**Comment by shishirbychapur** [comment]
> Pagination will be implemented subsequently by @yunruu

i thought session history is taken from matching db? why do we need pagination for collabDB?

**Comment by yunruu** [comment]
> > Pagination will be implemented subsequently by @yunruu
> 
> i thought session history is taken from matching db? why do we need pagination for collabDB?

Yep session history retrieval is being done on matching service

**Review by glemenneo — APPROVED** [review]

**Pull request merged**